lib/classifier-reborn/bayes.rb

```diff
@@ -73,7 +73,9 @@ def train(category, text)
 
       @backend.update_category_training_count(category, 1)
       @backend.update_total_trainings(1)
-      Hasher.word_hash(text, @language, @enable_stemmer).each do |word, count|
+      word_hash = Hasher.word_hash(text, @language, @enable_stemmer)
+      return if word_hash.length == 0
```

Comment:
Do you want to preempt `add_category` or just `update_category_training_count` and `update_total_trainings`?